src/main/java/uk/gov/laa/ccms/caab/service/AssessmentService.java

```diff
@@ -0,0 +1,559 @@
+              opponent.getAuditTrail().getLastSaved(),
+              assessment.getAuditDetail().getCreated())) {
+
+            log.info("Reassessment Required for {} as When individual is updated - IS TRUE",
+                application.getCaseReferenceNumber());
+            return true;
+          }
+        }
+
+        //When Individual/Organisation deleted
+        //used specifically for merits - this should be included for both assessments,
+        //although not specifically needed for means, both assessments and application data should
+        //be kept in sync.
+        final AssessmentEntityTypeDetail opponentEntityType =
+            getAssessmentEntityType(assessment, OPPONENT.getType());
+        if ((application.getOpponents() != null) && (opponentEntityType != null)) {
+          if (application.getOpponents().size() < opponentEntityType.getEntities().size()) {
+            log.info(
+                "Reassessment Required for {} as When organisation/individual is deleted "
+                    + "condition - IS TRUE", application.getCaseReferenceNumber());
+            return true;
+          }
+        }
+
+        if (application.getCostLimit() == null
+            || application.getCostLimit().getLimitAtTimeOfMerits() == null
+            || application.getCostLimit().getLimitAtTimeOfMerits()
+            .compareTo(application.getCosts().getRequestedCostLimitation()) < 0) {
+
+          log.info("Merit Reassessment Required for {} as app.getCostLimitAtTimeOfMerits() == null "
+                  + "|| app.getCostLimitAtTimeOfMerits().compareTo(app.getCosts()"
+                  + ".getRequestedOrDefaultCostLimitation()) < 0 IS TRUE",
+              application.getCaseReferenceNumber());
+          return true;
+        }
+
+      }
+    } else {
+      //todo amendment - need an EBS Case to workout if its required
+      //if (application.getAmendment()){
+      //  if (meansAssessment == null && !ebsCase.hasEbsAmendments()
+      //    && application.getApplicationType().getId().equals("SUBSTANTIVE")
+      //    && ebsCase.getCertificateType().getId().equals("EMERGENCY")
+      //  ){
+      //    return true;
+      //  }
+      //}
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if there are any discrepancies between the key data of the proceedings in the
+   * application and their corresponding assessment records based on the specified assessment
+   * entity type.
+   *
+   * @param application the application containing proceedings to check
+   * @param assessmentEntityType the type of assessment entity to match against proceedings
+   * @return true if any discrepancies are found; false otherwise
+   */
+  protected boolean checkAssessmentForProceedingKeyChange(
+      final ApplicationDetail application,
+      final AssessmentEntityTypeDetail assessmentEntityType) {
+
+    if (assessmentEntityType != null && application.getProceedings() != null) {
```

Comment:
Can we flip the condition here?
if (assessmentEntityType == null || application.getProceedings() == null) {
   return true;
}

then the rest of it doesn't need to be inside an else block

Reply:
Amended